_posts/2024-04-02-shared-conda-tutorial.md

```diff
@@ -0,0 +1,103 @@
+
+```bash
+# Create shared conda folder
+CONDA_PATH="/apps/conda"
+sudo mkdir -pv "${CONDA_PATH}"
+sudo chmod 755 "${CONDA_PATH}"
+
+# Download Miniconda installer
+CONDA_URL="https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh"
+wget "${CONDA_URL}" --output-document=/tmp/conda_installer.sh
+
+# Install Miniconda
+# -b          run install in batch mode (without manual intervention),
+#             it is expected the license terms (if any) are agreed upon
+# -u         update an existing installation (needed to install in existing dir)
+# -p PREFIX  install prefix, defaults to $PREFIX, must not contain spaces.
+sudo bash /tmp/conda_installer.sh -b -u -p "${CONDA_PATH}"
+
+# Set default conda settings for all users
+# (these will show up in $CONDA_PATH/.condarc)
+sudo "${CONDA_PATH}"/bin/conda config --system --set channel_priority strict
+sudo "${CONDA_PATH}"/bin/conda config --system --add channels defaults
+sudo "${CONDA_PATH}"/bin/conda config --system --add channels bioconda
+sudo "${CONDA_PATH}"/bin/conda config --system --add channels conda-forge
+```
+
+Users can override default conda settings in their own `$HOME/.condarc` configuration files.
+
+
+### Creating environments in the shared conda installation
+Any user with sudo privileges can create environments in the shared folder:
+```bash
+# activate base 
+source /apps/conda/bin/activate
+
+#create new environment
+sudo conda create -n myenv ...
+```
+
+### Using the shared conda installation
+Users can activate the shared conda base environment by running
+```bash
+/apps/conda/bin/conda init
+```
+to permanently initialize it for their user.
+This will add a section to the user's `$HOME/.bashrc`, which activates the shared conda base environment upon login.
+
+If users prefer not to automatically activate conda upon login, users can also opt to manually run 
+```bash
+source /apps/conda/bin/activate [optional_conda_env_name]
+```
+to temporarily activate it each time they want to use the shared conda installation (and use conda as usual for the rest of this bash session).
+
+
+### User-specific conda environments
+Users can create their own custom conda environments like usual, e.g. `conda create -n myenv ...`, but since users cannot write into the shared conda installation folder conda will automatically put the users' own environments in `$HOME/.conda/envs/`. Note that environments created by users this way are unavailable to other users on the system, unless they opened up the permissions of their home folder.
+
+
+## Other Tips & Tricks
+
+### Activating Conda Dynamically
+
+If you do not like having conda always active due to shell startup speed impact, or having to remember to `source ...` it, you can do like @hexylena does and source it dynamically on first invocation.
```

Comment:
It would be nice to link to @hexylena profil (GitHub or on the website) here. Wdyt?

Reply:
I think I'll copy in the auto-linking from the GTN, so, for now we can ignore this.